### ISSUE TITLE
[Core] task scheduling hangs forever

### ISSUE THREAD

**Issue by scv119** [ray-project/ray#19326]
### Search before asking

- [X] I searched the [issues](https://github.com/ray-project/ray/issues) and found no similar issues.

### Ray Component

Ray Core

### What happened + What you expected to happen

The dataset ingestion e2e test [hangs with following error](https://beta.anyscale.com/o/anyscale-internal/projects/prj_2xR6uT6t7jJuu1aCwWMsle/clusters/ses_rDUinK3MmFwBGecNKj3XnUi9). This could either be dataset bug/scheduling error/reporting error.
```
2021-10-12 05:44:32,010	WARNING worker.py:1228 -- The actor or task with ID 38b7b52b8a7371117241b1a3f60f62e6f940fd5479d1b53b cannot be scheduled right now. It requires {node:172.31.57.230: 0.001000}, {CPU: 1.000000} for placement, however the cluster currently cannot provide the requested resources. The required resources may be added as autoscaling takes place or placement groups are scheduled. Otherwise, consider reducing the resource requirements of the task.

```

another case:

https://anyscaleteam.slack.com/archives/CLJQR6420/p1634346237198100

and another one:

#19207

### Versions / Dependencies

latest ray

### Reproduction script

N/A

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

**Comment by scv119** [comment]
aha https://github.com/ray-project/ray/issues/19207 look similar!

**Comment by ericl** [comment]
Interesting! Looks like some issue with custom resources not getting synced, the commonality in both cases.

**Comment by scv119** [comment]
now the pipeline starts to failing in the hovorod

**Comment by scv119** [comment]
```
Required resources for this actor or task: {bundle_group_e3a58472ec387b230fb98f4b95d37cd9: 0.001000}
Available resources on this node: {7.000000/8.000000 CPU, 331721760.058594 GiB/331721760.058594 GiB memory, 165860879.980469 GiB/165860879.980469 GiB object_store_memory, 1.000000/1.000000 CPU_group_e3a58472ec387b230fb98f4b95d37cd9, 999.999000/1000.000000 bundle_group_e3a58472ec387b230fb98f4b95d37cd9, 1.000000/1.000000 CPU_group_0_e3a58472ec387b230fb98f4b95d37cd9, 1000.000000/1000.000000 bundle_group_0_e3a58472ec387b230fb98f4b95d37cd9, 1.000000/1.000000 node:127.0.0.1}
```

**Comment by rkooo567** [comment]
Potentially resolved after 
https://github.com/ray-project/ray/pull/19911
and 
https://github.com/ray-project/ray/pull/19823

We can reopen if we see this issue again
